TRW-1331 - Variable column widths

jasonhale opened the pull request:
- Added variable column widths to the Details table while respecting 100% table width.
- Converted sticky table header to use CSS, removed the need for a JS dependency.

troyblank commented on src/sass/partials/_data-details.scss:
```diff
@@ -33,13 +37,6 @@
 
     tbody {
-        // in order to calculate height for this dynamically
-        // based on parent you need sticky-header-table.js
```
can we remove `sticky-header-table.js` here?

jasonhale replied:
Ah.  Yeah, we could.  Which shows me that I need to update the demo.  I'll get on that.

troyblank commented:
Nit picky but "Variable column widths" could be a tad bit more descriptive. Given I know what you are doing I get it but I am not sure we would understand it was for the details table if we were scanning for a commit in the future.

jasonhale commented:
@troyblank - removed the stickyHeaderTable.js and references, updated the demo files, and fixed a few browser-specific issues (like, `position:sticky` doesn't work on `<thead>` in Chrome... assigned to `<th>` instead).

troyblank commented on src/js/tiles/stickyHeaderTable.js:
```diff
@@ -1,27 +0,0 @@
-import dom from '../util/dom';
```
bye bye!
